packages/chrono-core/src/chrono.ts

```diff
@@ -8,8 +8,16 @@ export type ScheduleTaskInput<TaskKind, TaskData, DatastoreOptions> = ScheduleIn
   DatastoreOptions
 >;
 
-export class Chrono<TaskKind, DatastoreOptions> extends EventEmitter {
+export type RegisterTaskHandlerInput<TaskKind, TaskData> = {
+  kind: TaskKind;
+  handler: (task: Task<TaskKind, TaskData>) => Promise<void>;
+};
+
+export class Chrono<TaskMapping extends Record<PropertyKey, unknown>, DatastoreOptions> extends EventEmitter {
   #datastore: Datastore<DatastoreOptions>;
+  #handlers: Partial<{
+    [Key in keyof TaskMapping]: (task: Task<Key, TaskMapping[Key]>) => Promise<void>;
+  }> = {};
```

Comment:
could not for the life of my figure out how to get this working with a Map. Will try again later but this works well for now